platform/server-bundle/src/main/resources/webapp/js/services.js

```diff
@@ -2,9 +2,274 @@
+                callback: callback && callback !== undefined ? callback : null
+            });
+        };
+
+        modalFactory.motechAlertStackTrace = function (msg, title, response, callback) {
+            if (title === null || title === '') {
+                title = 'server.bootstrapDialog.alert';
+            }
+            return modalFactory.alert({
+                type: 'type-danger', //Error type
+                title: jQuery.i18n.prop(title),
+                message: jQuery.i18n.prop(msg).bold() + ": \n" + response,
+                callback: callback
+            });
+        };
+
+        modalFactory.confirm = function (msg, title, callback) {
+            if (typeof msg === 'object' && msg.constructor === {}.constructor) {
+                return makeConfirm(msg);
+            } else {
+                return makeConfirm({
+                    message: jQuery.i18n.prop(msg),
+                    title: title && title !== undefined ? jQuery.i18n.prop(title) : jQuery.i18n.prop('server.bootstrapDialog.confirm'),
+                    callback: callback && callback !== undefined ? callback : null
+                });
+            }
+        };
+
+        modalFactory.errorAlert = function (response) {
+            LoadingModal.close();
+            return modalFactory.alert({
+                type: 'type-danger', //Error type
+                title: jQuery.i18n.prop('server.error'),
+                message: response.status + ": " + response.statusText
+            });
+        };
+
+        modalFactory.handleResponse = function (title, defaultMsg, response, callback) {
+            var msg = { value: "server.error", literal: false, params: [] },
+                responseData = (typeof(response) === 'string') ? response : response.data;
+
+            LoadingModal.close();
+            msg = parseResponse(responseData, defaultMsg);
+
+            if (callback) {
+                callback(title, msg.value, msg.params);
+            } else if (msg.literal) {
+                return modalFactory.alert({
+                    type: 'type-danger',
+                    title: jQuery.i18n.prop(title),
+                    message: msg.value,
+                    callback: callback
+                });
+            } else {
+                return modalFactory.motechAlert(msg.value, title, msg.params);
+            }
+        };
+
+        modalFactory.alertHandler = function (msg, title, callback) {
+            return function() {
+                LoadingModal.close();
+                return modalFactory.alert(msg, title, callback);
+            };
+        };
+
+        modalFactory.angularHandler = function(title, defaultMsg, callback) {
+            return function(response) {
+                return modalFactory.handleResponse(title, defaultMsg, response, callback);
+            };
+        };
+
+        modalFactory.handleWithStackTrace = function(title, defaultMsg, response) {
+            var msg = "server.error";
```

Comment:
Isn't the msg variable actually the defaultMsg, and the defaultMsg parameter the msg argument (which motechAlert is the first argument)